### PR TITLE
[AutoDiff] Fix a subset-parameters thunk parameter indexing crasher.

### PR DESCRIPTION
**Pull request by rxwei**
The subset parameters thunk generation logic is not handling cases where the original JVP/JVP is already producing a subset-parameters linear map. This patch fixes it by computing indices w.r.t. the linear map from indices w.r.t. the original function. 

Resolves [TF-594](https://bugs.swift.org/browse/TF-594) and unblocks tensorflow/swift-apis#147.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Review by dan-zheng — APPROVED** [review]

File: test/AutoDiff/superset_adjoint.swift [hunk 1]
```diff
@@ -33,6 +33,14 @@ SupersetVJPTests.test("CrossModuleClosure") {
   expectEqual(1, gradient(at: Float(1)) { x in x + 2 })
 }
 
+SupersetVJPTests.test("SubsetOfSubset") {
+  @differentiable(wrt: (x, z))
+  func foo(_ x: Float, _ y: Float, _ z: Float) -> Float {
+    Float(0).withoutDerivative()
```
Comment (eaplatanios) [inline test/AutoDiff/superset_adjoint.swift]: `Differentiable.withoutDerivative()` has now been removed. Shouldn't this be `withoutDerivative(at: Float(0))` instead?

Comment (rxwei) [reply test/AutoDiff/superset_adjoint.swift]: Good catch!

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Pull request merged**